File: Algorithm/implement_stack_using_queues.py
import collections
class Stack:
    # initialize your data structure here.
    def __init__(self):
        self.queue = collections.deque()

    # @param x, an integer
    # @return nothing
    def push(self, x):
        n = len(self.queue)
        while n != 0 :
            self.queue.appendleft(self.queue.pop())
            n -= 1
        self.queue.appendleft(x)
    # @return nothing
    def pop(self):
        print self.queue
        return self.queue.popleft()

    # @return an integer
    def top(self):
        return self.queue[0]       
    # @return an boolean
    def empty(self):
        return len(self.queue) == 0

if __name__ == '__main__':

    Stack = Stack()
    Stack.push(1)

    print Stack.pop()
    Stack.push(5)
    print Stack.top()
    

File: Algorithm/CPP/implement_stack_using_queues.cpp
# include <iostream>
# include <string>
using namespace std;



class Stack {
public:
	// Push element x onto stack.
	void push(int x) {
		Queue1.push(x);

	}

	// Removes the element on top of the stack.
	void pop() {
		int count = 0;
		while (!Queue1.empty())
		{
			Queue2.push(Queue1.front());
			Queue1.pop();
			count++;
		}
		while (--count)
		{
			Queue1.push(Queue2.front());
			Queue2.pop();
		}
		Queue2.pop();
	}

	// Get the top element.
	int top() {
		int count = 0;
		while (!Queue1.empty())
		{
			Queue2.push(Queue1.front());
			Queue1.pop();
			count++;
		}
		while (--count)
		{
			Queue1.push(Queue2.front());
			Queue2.pop();
		}
		int top = Queue2.front();
		Queue2.pop();
		Queue1.push(top);
		return top;

	}

	// Return whether the stack is empty.
	bool empty() {
		return Queue1.empty();
	}

private:
	queue<int> Queue1;
	queue<int> Queue2;
};
